dataset/src/main/java/org/kitesdk/examples/data/GenerateEvents.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015 Cloudera, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. 
+ */
+
+package org.kitesdk.examples.data;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import org.apache.avro.util.Utf8;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.util.ToolRunner;
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.DatasetWriter;
+import org.kitesdk.data.Datasets;
+import org.kitesdk.data.View;
+import org.kitesdk.data.event.StandardEvent;
+
+public class GenerateEvents extends BaseEventsTool {
+  protected Random random;
+  protected long baseTimestamp;
+  protected long counter;
+
+  public GenerateEvents() {
+    random = new Random();
+    baseTimestamp = System.currentTimeMillis();
+    counter = 0l;
+  }
+
+  @Override
+  public int run(List<String> args) throws Exception {
+	  return 0;
+  }
+  
+  public int run(String[] args) throws Exception {
+
+    View<StandardEvent> events = Datasets.load(
+        "dataset:hive:events", StandardEvent.class);
+
+    DatasetWriter<StandardEvent> writer = events.newWriter();
+    try {
+      Utf8 sessionId = new Utf8("sessionId");
+      long userId = 0;
+      Utf8 ip = new Utf8("ip");
+      int randomEventCount = 0;
+      while (System.currentTimeMillis() - baseTimestamp < 36000) {
+        sessionId = randomSessionId();
+        userId = randomUserId();
+        ip = randomIp();
+        randomEventCount = random.nextInt(25);
+        for (int i=0; i < randomEventCount; i++) {
+            writer.write(generateRandomEvent(sessionId, userId, ip));
+        }
+      }
+    } finally {
+      writer.close();
+    }
+
+    System.out.println("Generated " + counter + " events");
+
+    return 0;
+  }
+
+  public StandardEvent generateRandomEvent(Utf8 sessionId, long userId, Utf8 ip) {
+    return StandardEvent.newBuilder()
+        .setEventInitiator(new Utf8("client_user"))
+        .setEventName(randomEventName())
+        .setUserId(userId)
+        .setSessionId(sessionId)
+        .setIp(ip)
+        .setTimestamp(randomTimestamp())
+        .build();
+  }
+
+  public Utf8 randomEventName() {
+    return new Utf8("event"+counter++);
```

Comment:
Since this is no longer random, it would make more sense to move it into the loop in run and keep track of the counter as a local variable.